src/arcade/core/util/Plane.java

```diff
@@ -0,0 +1,82 @@
+package arcade.core.util;
+
+import sim.util.Int3D;
+
+/**
+ * A plane in 3D space.
+ */
+
+public final class Plane {
+    /** A point on the plane. */
+    public final Int3D point;
+
+    /** The normal vector to the plane. */
+    public final Int3D normalVector;
+
+    /**
+     * Creates a plane from a point and a vector.
+     *
+     * @param point  a point on the plane
+     * @param normalVector  the normal vector to the plane
+     */
+    public Plane(Int3D point, Int3D normalVector) {
+        this.point = point;
+        this.normalVector = normalVector;
+    }
+
+    /**
+     * Determines whether a point is on the plane.
+     *
+     * @param p  the point to test
+     * @return  {@code true} if the point is on the plane
+     *        {@code false} otherwise
+     */
+    public boolean isOnPlane(Int3D p) {
```

Comment:
Is this used anywhere? I'd also replace the calculate with a call to `distanceToPlane` rather than having the same calculation twice.